xarray/testing.py

```diff
@@ -129,8 +129,9 @@ def assert_allclose(a, b, rtol=1e-05, atol=1e-08, decode_bytes=True):
             # DataArray, so call into _data_allclose_or_equiv directly
             allclose = _data_allclose_or_equiv(a.coords[v].values,
                                                b.coords[v].values, **kwargs)
-            assert allclose, '{}\n{}'.format(a.coords[v].values,
-                                             b.coords[v].values)
+            assert allclose, 'coord {}:{}\n{}'.format(a.coords[v].name,
+                                                      a.coords[v].values,
+                                                      b.coords[v].values)
```

Comment:
What about adding `name` to the error message for non-coordinates, too? We could do this by adding an optional `name` argument to this function. (It would also be useful for `assert_equal` and `assert_identical`.)